### PR TITLE
Improved test coverage for policy in pkg/detector

### PR DESCRIPTION
**Pull request by NishantBansal2003**
**What type of PR is this?**
/kind feature
/kind failing-test

**What this PR does / why we need it**:
The test case coverage for pkg/detector/policy.go has been increased upto 50%.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
To verify the changes in the pkg/detector/ directory run the following commands:
```
go test ./... -coverprofile=coverage.out
go tool cover -html=coverage.out -o coverage.html
open coverage.html
```
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by yashpandey06** [comment]
@NishantBansal2003 Can you please just mention the referenced issue here in the  pr description for more visibility. Rather mention it as a part of or fixes of .

**Comment by NishantBansal2003** [comment]
PTAL...
cc: @XiShanYongYe-Chang

File: pkg/detector/policy_test.go [hunk 1]
```diff
@@ -37,6 +37,829 @@ import (
...
+			resourceDetector := &ResourceDetector{
+				Client:          fakeClient,
+				DynamicClient:   fakeDynamicClient,
+				RESTMapper:      fakeClient.RESTMapper(),
+				InformerManager: genMgr,
+			}
+			err := resourceDetector.cleanPPUnmatchedRBs(tt.policyID, tt.policyNamespace, tt.policyName, tt.selectors)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("cleanPPUnmatchedRBs() error = %v, wantErr %v", err, tt.wantErr)
+			}
```
Comment (XiShanYongYe-Chang) [inline pkg/detector/policy_test.go:162]: In addition to determining whether err is returned, can we add more checks to determine whether the target object has been cleaned up?

**Comment by XiShanYongYe-Chang** [comment]
Thanks
/assign

File: pkg/detector/policy_test.go (hunk 1 above)
Comment (NishantBansal2003) [reply pkg/detector/policy_test.go:162]: Sure, let me check...

**Comment by NishantBansal2003** [comment]
Made the required changes, PTAL
cc: @XiShanYongYe-Chang

**Comment by NishantBansal2003** [comment]
Could you please review this, also? Thank you!
cc: @XiShanYongYe-Chang

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks
/lgtm
/approve

**Pull request merged**